File: samples/samples-cs/HomeSeerSamplePlugin/Constants/Devices.cs
using System.Collections.Generic;

namespace HSPI_HomeSeerSamplePlugin.Constants {

    public static class Devices {
        
        public static List<string> SampleDeviceTypeList => new List<string> {
                                                                    "Line-powered switch",
                                                                    "Line-powered sensor",
                                                                    "Battery-powered sensor",
                                                                    "Thermostat"
                                                                };
        
        public static List<string[]> SampleDeviceTypeFeatures => new List<string[]>
                                                                     {
                                                                         LinePoweredSwitchFeatures,
                                                                         LinePoweredSensorFeatures,
                                                                         BatteryPoweredSensorFeatures,
                                                                         ThermostatFeatures
                                                                     };
        
        public static string[] LinePoweredSwitchFeatures => new []{ "On-Off control feature" };
        public static string[] LinePoweredSensorFeatures => new []{ "Open-Close status feature" };
        public static string[] BatteryPoweredSensorFeatures => new[] { "Open-Close status feature", "Battery status feature" };
        public static string[] ThermostatFeatures => new[] { "Ambient Temperature feature", 
                                                             "Heating Setpoint feature",
                                                             "Cooling Setpoint feature",
                                                             "HVAC Mode feature",
                                                             "HVAC Status feature",
                                                             "Fan Mode feature",
                                                             "Fan Status feature",
                                                             "Humidity feature"
                                                            };

        public const int ThermostatHvacModeHeat = 1;
        public const int ThermostatHvacModeCool = 2;
        public const int ThermostatHvacModeAuto = 3;
        public const int ThermostatHvacModeAuxHeat = 4;
        public const int ThermostatHvacModeOff = 5;

        public const int ThermostatHvacStatusIdle = 0;
        public const int ThermostatHvacStatusHeating = 1;
        public const int ThermostatHvacStatusCooling = 2;

        public const int ThermostatFanModeAuto = 1;
        public const int ThermostatFanModeOn = 2;

        public const int ThermostatFanStatusOff = 0;
        public const int ThermostatFanStatusOn = 1;

        public const int ThermostatSetpointDecrement = 1000;
        public const int ThermostatSetpointIncrement = 1001;

        public const string DeviceConfigPageId = "device-config-page";
        public const string DeviceConfigPageName = "Sample Device Config";

        public static string DeviceConfigLabelWTitleId => $"{DeviceConfigPageId}-samplelabel1";
        public const string DeviceConfigLabelWTitleName = "Sample label with title";
        public const string DeviceConfigLabelWTitleValue = "This is a label with a title";
        public static string DeviceConfigLabelWoTitleId => $"{DeviceConfigPageId}-samplelabel2";
        public const string DeviceConfigLabelWoTitleValue = "This is a label without a title";
        public static string DeviceConfigSampleToggleId => $"{DeviceConfigPageId}-sampletoggle1";
        public const string DeviceConfigSampleToggleName = "Sample Toggle Switch";
        public static string DeviceConfigSampleCheckBoxId => $"{DeviceConfigPageId}-samplecheckbox1";
        public const string DeviceConfigSampleCheckBoxName = "Sample Checkbox";
        public static string DeviceConfigSelectListId => $"{DeviceConfigPageId}-sampleselectlist1";
        public const string DeviceConfigSelectListName = "Sample Dropdown Select List";
        public static string DeviceConfigRadioSlId => $"{DeviceConfigPageId}-sampleselectlist2";
        public const string DeviceConfigRadioSlName = "Sample Radio Select List";
        public static List<string> DeviceConfigSelectListOptions => new List<string> {
                                                                                "Option 1",
                                                                                "Option 2",
                                                                                "Option 3"
                                                                            };
        public static string DeviceConfigInputId => $"{DeviceConfigPageId}-sampleinput";
        public const string DeviceConfigInputName = "Sample Text Input";
        public const string DeviceConfigInputValue = "This is a text input";
        public static string DeviceConfigDateInputId => $"{DeviceConfigPageId}-sampledateinput";
        public const string DeviceConfigDateInputName = "Sample Date Input";
        public const string DeviceConfigDateInputValue = "";
        public static string DeviceConfigTimeInputId => $"{DeviceConfigPageId}-sampletimeinput";
        public const string DeviceConfigTimeInputName = "Sample Time Input";
        public const string DeviceConfigTimeInputValue = "";

        public static string DeviceConfigTextAreaId => $"{DeviceConfigPageId}-sampletextarea";
        public const string DeviceConfigTextAreaName = "Sample Text Area";
        
        public static string DeviceConfigTimeSpanId => $"{DeviceConfigPageId}-sampletimespan";
        public const string DeviceConfigTimeSpanName = "Sample Time Span";

        public static string DeviceConfigNavButton1Id => $"{DeviceConfigPageId}-samplenavbutton1";
        public static string DeviceConfigNavButton2Id => $"{DeviceConfigPageId}-samplenavbutton2";

    }

}

File: samples/samples-vb/HomeSeerSamplePlugin/Constants/Devices.vb
Public Module Devices
    
    Public ReadOnly Property SampleDeviceTypeList As List(Of String)
        Get
            Return New List(Of String) From {
                "Line-powered switch",
                "Line-powered sensor",
                "Battery-powered sensor",
                "Thermostat"
                }
        End Get
    End Property

    Public ReadOnly Property SampleDeviceTypeFeatures As List(Of String())
        Get
            Return New List(Of String()) From {
                LinePoweredSwitchFeatures,
                LinePoweredSensorFeatures,
                BatteryPoweredSensorFeatures,
                ThermostatFeatures
                }
        End Get
    End Property

    Public ReadOnly Property LinePoweredSwitchFeatures As String()
        Get
            Return {"On-Off control feature"}
        End Get
    End Property

    Public ReadOnly Property LinePoweredSensorFeatures As String()
        Get
            Return {"Open-Close status feature"}
        End Get
    End Property

    Public ReadOnly Property BatteryPoweredSensorFeatures As String()
        Get
            Return {"Open-Close status feature", "Battery status feature"}
        End Get
    End Property

    Public ReadOnly Property ThermostatFeatures As String()
        Get
            Return {"Ambient Temperature feature", "Heating Setpoint feature", "Cooling Setpoint feature", "HVAC Mode feature", "HVAC Status feature", "Fan Mode feature", "Fan Status feature", "Humidity feature"}
        End Get
    End Property

    Public Const ThermostatHvacModeHeat As Integer = 1
    Public Const ThermostatHvacModeCool As Integer = 2
    Public Const ThermostatHvacModeAuto As Integer = 3
    Public Const ThermostatHvacModeAuxHeat As Integer = 4
    Public Const ThermostatHvacModeOff As Integer = 5
    Public Const ThermostatHvacStatusIdle As Integer = 0
    Public Const ThermostatHvacStatusHeating As Integer = 1
    Public Const ThermostatHvacStatusCooling As Integer = 2
    Public Const ThermostatFanModeAuto As Integer = 1
    Public Const ThermostatFanModeOn As Integer = 2
    Public Const ThermostatFanStatusOff As Integer = 0
    Public Const ThermostatFanStatusOn As Integer = 1
    Public Const ThermostatSetpointDecrement As Integer = 1000
    Public Const ThermostatSetpointIncrement As Integer = 1001

    Public Const DeviceConfigPageId As String = "device-config-page"
    Public Const DeviceConfigPageName As String = "Sample Device Config"

    Public ReadOnly Property DeviceConfigLabelWTitleId As String
        Get
            Return $"{DeviceConfigPageId}-samplelabel1"
        End Get
    End Property

    Public Const DeviceConfigLabelWTitleName As String = "Sample label with title"
    Public Const DeviceConfigLabelWTitleValue As String = "This is a label with a title"

    Public ReadOnly Property DeviceConfigLabelWoTitleId As String
        Get
            Return $"{DeviceConfigPageId}-samplelabel2"
        End Get
    End Property

    Public Const DeviceConfigLabelWoTitleValue As String = "This is a label without a title"

    Public ReadOnly Property DeviceConfigSampleToggleId As String
        Get
            Return $"{DeviceConfigPageId}-sampletoggle1"
        End Get
    End Property

    Public Const DeviceConfigSampleToggleName As String = "Sample Toggle Switch"

    Public ReadOnly Property DeviceConfigSampleCheckBoxId As String
        Get
            Return $"{DeviceConfigPageId}-samplecheckbox1"
        End Get
    End Property

    Public Const DeviceConfigSampleCheckBoxName As String = "Sample Checkbox"

    Public ReadOnly Property DeviceConfigSelectListId As String
        Get
            Return $"{DeviceConfigPageId}-sampleselectlist1"
        End Get
    End Property

    Public Const DeviceConfigSelectListName As String = "Sample Dropdown Select List"

    Public ReadOnly Property DeviceConfigRadioSlId As String
        Get
            Return $"{DeviceConfigPageId}-sampleselectlist2"
        End Get
    End Property

    Public Const DeviceConfigRadioSlName As String = "Sample Radio Select List"

    Public ReadOnly Property DeviceConfigSelectListOptions As List(Of String)
        Get
            Return New List(Of String) From {
                "Option 1",
                "Option 2",
                "Option 3"
            }
        End Get
    End Property

    Public ReadOnly Property DeviceConfigInputId As String
        Get
            Return $"{DeviceConfigPageId}-sampleinput"
        End Get
    End Property

    Public Const DeviceConfigInputName As String = "Sample Text Input"
    Public Const DeviceConfigInputValue As String = "This is a text input"

    Public ReadOnly Property DeviceConfigDateInputId As String
        Get
            Return $"{DeviceConfigPageId}-sampledateinput"
        End Get
    End Property

    Public Const DeviceConfigDateInputName As String = "Sample Date Input"
    Public Const DeviceConfigDateInputValue As String = ""

    Public ReadOnly Property DeviceConfigTimeInputId As String
        Get
            Return $"{DeviceConfigPageId}-sampletimeinput"
        End Get
    End Property

    Public Const DeviceConfigTimeInputName As String = "Sample Time Input"
    Public Const DeviceConfigTimeInputValue As String = ""

    Public ReadOnly Property DeviceConfigTextAreaId As String
        Get
            Return $"{DeviceConfigPageId}-sampletextarea"
        End Get
    End Property

    Public Const DeviceConfigTextAreaName As String = "Sample Text Area"

    Public ReadOnly Property DeviceConfigTimeSpanId As String
        Get
            Return $"{DeviceConfigPageId}-sampletimespan"
        End Get
    End Property

    Public Const DeviceConfigTimeSpanName As String = "Sample Time Span"

    Public ReadOnly Property DeviceConfigNavButton1Id As String
        Get
            Return $"{DeviceConfigPageId}-samplenavbutton1"
        End Get
    End Property
    Public ReadOnly Property DeviceConfigNavButton2Id As String
        Get
            Return $"{DeviceConfigPageId}-samplenavbutton2"
        End Get
    End Property
End Module